src/dataframe/io.jl

```diff
@@ -848,7 +846,13 @@ function readtable(io::IO,
                      allowescapes, normalizenames)
 
     # Use the IO stream method for readtable()
-    df = readtable!(p, io, nrows, o)
+    if encoding == "UTF-8"
+        df = readtable!(p, io, nrows, o)
+    else # we wrap the IO stream in a decoder
+        d_io = StringEncodings.StringDecoder(io, encoding)
+        df = readtable!(p, d_io, nrows, o)
```

Comment:
It would be good to use `try.. finally` to be sure the handle is closed even in case of error.